### ISSUE TITLE
Remove -none and -any version range syntax.

### ISSUE THREAD

**Issue by phadej** [haskell/cabal#6589]
They make version grammar non-declarative.

If we have grammar for `Dependency` as

    PKGNAME OPTSPACE VERSIONRANGE

then valid package name `foo` and version range `-any && >= 1` can be combined into

    foo-any && >= 1

which isn't valid.

Yet, `foo>=1` is valid (and often used).

Alternatively we can specify that space is mandatory before `-any` and `-none` tokens, that will make grammar less compositional.

---

TLDR; make version-range syntax so it cannot be interpretted as part of `UnqualComponentName`. Simplest solution is to require `VersionRange` to start with character not allowed in `UnqualComponentName`.

**Comment by phadej** [comment]
Another reason to remove `-any` and `-none` is that they clash with constraint syntax:

```cabal
constraints: foo -none
```

is that flag or version range? I'm not sure without trying.

**Comment by hvr** [comment]
I can't think of a *good* reason to keep `-none` or `-any` (even though I've certainly used `-none` in cabal.project/freeze files); the only reason I could think of is if we could give `-none` or `-any` some additional connotation beyond the semantics of `<0` and `>=0` that would make a difference for linters (such as `cabal check`) to signal intent; but then again I don't have a good use-case for that OTTOMH.
